airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -38,41 +39,32 @@ def connector_spec_dict_fixture(actual_connector_spec):
     return json.loads(actual_connector_spec.json())
 
 
-@pytest.fixture(name="actual_connector_spec")
-def actual_connector_spec_fixture(request: BaseTest, docker_runner: ConnectorRunner) -> ConnectorSpecification:
-    if not request.instance.spec_cache:
-        output = docker_runner.call_spec()
-        spec_messages = filter_output(output, Type.SPEC)
-        assert len(spec_messages) == 1, "Spec message should be emitted exactly once"
-        spec = spec_messages[0].spec
-        request.spec_cache = spec
-    return request.spec_cache
-
-
-@pytest.fixture(name="previous_connector_spec")
-def previous_connector_spec_fixture(
-    request: BaseTest, previous_connector_docker_runner: ConnectorRunner
-) -> Optional[ConnectorSpecification]:
-    if previous_connector_docker_runner is None:
-        logging.warning(
-            "\n We could not retrieve the previous connector spec as a connector runner for the previous connector version could not be instantiated."
-        )
-        return None
-    if not request.instance.previous_spec_cache:
-        output = previous_connector_docker_runner.call_spec()
-        spec_messages = filter_output(output, Type.SPEC)
-        assert len(spec_messages) == 1, "Spec message should be emitted exactly once"
-        spec = spec_messages[0].spec
-        request.instance.previous_spec_cache = spec
-    return request.instance.previous_spec_cache
-
-
 @pytest.mark.default_timeout(10)
 class TestSpec(BaseTest):
 
     spec_cache: ConnectorSpecification = None
     previous_spec_cache: ConnectorSpecification = None
 
+    @staticmethod
+    def compute_spec_diff(actual_connector_spec: ConnectorSpecification, previous_connector_spec: ConnectorSpecification):
+        return DeepDiff(previous_connector_spec.dict(), actual_connector_spec.dict(), view="tree")
+
+    @pytest.fixture(name="spec_diff")
+    def spec_diff_fixture(
+        self, inputs: SpecTestConfig, actual_connector_spec: ConnectorSpecification, previous_connector_spec: ConnectorSpecification
+    ) -> DeepDiff:
+        if (
+            inputs.backward_compatibility_tests_config.disable_backward_compatibility_tests_for_version
+            == inputs.backward_compatibility_tests_config.previous_connector_version
```

Comment:
There's something interesting here with using `latest` as the previous version by default and checking this based on `previous_connector_version` directly... if we want to skip against version 1.2.3 but we want to keep testing against whatever the latest version is in the future, the skip wouldn't really work (`"1.2.3" != "latest")

I think we may need to get the actual version used in latest to make this work correctly. The version set within the image should be available as a [label on the Image object](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.Image.labels) as `io.airbyte.version`.

Reply:
Yep good catch!

Reply:
Here's my suggestion:
* If `previous_connector_version` == `latest`: retrieve the real airbyte version from the image label 
* Add validation on `disable_backward_compatibility_tests_for_version` to ensure a semver string is passed, disallowing `latest` value: it serves the purpose of skipping tests for a very specific version.

Reply:
I implemented this in: 
c6b1b0755c027d329e7d7dcc0e800fdd1cc61923
0ce08e509971c401353f4034a134da3112f90ff2